polyfill.js

```diff
@@ -11,27 +11,20 @@ function determineLevel(el) {
 }
 
 function adjustHeading(heading) {
-  let type = heading.localName,
-      parent = heading.parentNode;
-  if ((type === "h1" ||
-       type === "h2" ||
-       type === "h3" ||
-       type === "h4" ||
-       type === "h5" ||
-       type === "h6") &&
-      parent !== null && parent.localName === "hgroup") {
-    heading.setAttribute("role", "presentation");
-    heading.removeAttribute("aria-level");
-  } else if (type === "hgroup" || type === "h1") {
-    heading.setAttribute("aria-level", determineLevel(heading));
-    if (type === "hgroup") {
-      heading.setAttribute("role", "heading");
-    }
+  let type = heading.localName;
+  if (type === "hgroup") {
+    heading.setAttribute("role", "heading");
+    heading.querySelectorAll("h1,h2,h3,h4,h5,h6").forEach(h => {
+      h.setAttribute("role", "presentation");
+      h.removeAttribute("aria-level");
+    })
```

Comment:
Thank you, but this ends up being slightly different as the original code only looked at a parent hgroup, whereas this looks at an ancestor hgroup (by querying descendants rather than only children). And that's probably not what we want to require from browser implementations as that would be a bit more expensive.

Reply:
Ah! That is a great point and I did not catch this difference. Thanks for the explanation.